lib/consumer.ex

```diff
@@ -397,13 +401,20 @@ defmodule GenRMQ.Consumer do
         queue = config[:queue]
         exchange = config[:exchange]
         dl_queue = config[:deadletter_queue] || "#{queue}_error"
-        dl_exchange = config[:deadletter_queue] || "#{exchange}.deadletter"
+        dl_exchange = config[:deadletter_exchange] || "#{exchange}.deadletter"
+        dl_routing_key = config[:deadletter_routing_key] || "#"
 
         Queue.declare(chan, dl_queue, durable: true, arguments: setup_ttl([], ttl))
         Exchange.topic(chan, dl_exchange, durable: true)
-        Queue.bind(chan, dl_queue, dl_exchange, routing_key: "#")
-
-        [{"x-dead-letter-exchange", :longstr, dl_exchange}]
+        Queue.bind(chan, dl_queue, dl_exchange, routing_key: dl_routing_key)
+
+        [{"x-dead-letter-exchange", :longstr, dl_exchange}] ++
+        case dl_routing_key do
```

Comment:
Do we need this case? Wouldn't always using `[{"x-dead-letter-routing-key", :longstr, dl_routing_key}]` work?

Reply:
My concern is with the bindings already in production because this option wasn't informed before and will cause a binding error.

Reply:
Very good point, thanks for clarifying. 👍